src/createSelector.js

```diff
@@ -131,23 +130,24 @@ const getInstanceSelector = (
   }
 
   result.clearCache = (recursive = true) => {
-    cache.clear()
-    usages.clear()
+    for (let key in cache) delete cache[key]
+    usages = {}
```

Comment:
Nope, they aren't being used in the same way.

`usages` is only used from inside `getInstanceSelector` and `cache` is being passed to `getComputeFn` which is responsible for updating/initializing the values of the `cache`... If I reset the cache to a new instance, then the `computeFn` will still use the old cache.

I have 2 options for sharing the cache with the computeFn: 
1) Don't pass a reference to `getComputeFn` and pass a "getter" function that would return the instance of the cache that's relevant  (what I've been doing until this PR)
2) Pass the reference directly and always use the same reference.

I went with option 2 because the output size was smaller and because perf-wise should be slightly better, because in reality the `clearCache` function won't be used at all or very infrequently.

Reply:
On a second thought... I went back to using the getter, it's cleaner and in the way that I'm doing it now, the size it's actually smaller. Thanks for pointing that out! :+1: